packages/wallets/src/services/wallet-factory.ts

```diff
@@ -53,25 +76,56 @@ export class WalletFactory {
                 },
                 linkedUser,
             });
-            this.assertCorrectWalletType(walletResponse, "solana-smart-wallet");
+        } else if (type === "solana-mpc-wallet") {
+            const { linkedUser } = args as WalletTypeToArgs["solana-mpc-wallet"];
+            return await this.apiClient.createWallet({
+                type: "solana-mpc-wallet",
+                linkedUser,
+            });
+        } else {
+            throw new Error("Not implemented");
+        }
+    }
+
+    private createWalletInstance<WalletType extends keyof WalletTypeToArgs>(
+        type: WalletType,
+        walletResponse: CreateWalletResponse,
+        args: WalletTypeToArgs[WalletType],
+        options?: WalletOptions
+    ): WalletTypeToWallet[WalletType] {
+        this.assertCorrectWalletType(walletResponse, type);
+
+        if (type === "evm-smart-wallet") {
+            const { chain, adminSigner } = args as WalletTypeToArgs["evm-smart-wallet"];
```

Comment:
There's a bit of repeated code here and also this method ends up being way too long already, and will be even more once we support the other 5 wallet types we have. Can we abstract it?

Reply:
Made a light refactor to handle much more VMs in the future easily!

https://github.com/Crossmint/crossmint-sdk/pull/985/commits/74f7f614690aca8bccef2531d3ea43e9e53ec00b